### PR TITLE
Add config system to allow pride quilts even if it's not pride month

### PR DESCRIPTION
**Pull request by triphora**
This adds a simple config system by the name of MidnightConfig, written by Motschen.
It will, of course, still show the pride quilts during June, but this allows the user to have the ability to change it to show pride quilts manually even if it's not June.

### CODE REVIEW

**Comment by sylv256** [comment]
i have a mixin mod for this, though it'd be nice if there was a button for this in the config menu

**Comment by triphora** [comment]
@TehcJS There is one (that's why I added Mod Menu as a modImplementation). I forgot to include a screenshot
image

**Comment by sylv256** [comment]
oh lol

File: src/main/java/eu/midnightdust/lib/config/MidnightConfig.java [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+        Function<Object, Text> func = value -> new TranslatableText(modid + ".midnightconfig." + "enum." + type.getSimpleName() + "." + info.value.toString());
+        info.widget = new AbstractMap.SimpleEntry<ButtonWidget.PressAction, Function<Object, Text>>(button -> {
+          int index = values.indexOf(info.value) + 1;
+          info.value = values.get(index >= values.size() ? 0 : index);
+          button.setMessage(func.apply(info.value));
+        }, func);
+      }
+    }
+    entries.add(info);
+  }
```
Comment (sylv256) [inline src/main/java/eu/midnightdust/lib/config/MidnightConfig.java:172]: what's all of this reflection for?

Comment (sylv256) [reply src/main/java/eu/midnightdust/lib/config/MidnightConfig.java:172]: why do you need reflection*

Comment (triphora) [reply src/main/java/eu/midnightdust/lib/config/MidnightConfig.java:172]: I dunno, I didn't write it, it's copied from [here](https://github.com/TeamMidnightDust/MidnightLib/blob/main/src/main/java/eu/midnightdust/lib/config/MidnightConfig.java) which is alright to copy thanks to MIT

**Comment by darkerbit** [comment]
Does the config system used depend on loader internals or Fabric API in any way?

**Comment by triphora** [comment]
> Does the config system used depend on loader internals or Fabric API in any way?

Not really. I think it only relies on it for determining config path

**Comment by darkerbit** [comment]
Alright, I'll take a closer look in the morning then! :+1:

File: src/main/java/eu/midnightdust/lib/config/MidnightConfig.java (hunk 1 above)
Comment (sylv256) [reply src/main/java/eu/midnightdust/lib/config/MidnightConfig.java:172]: that'll be very slow. i'm sure there's a better way to do it

File: src/main/resources/assets/quilt-loading-screen/lang/en_us.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "quilt-loading-screen.midnightconfig.title": "\u00a7lQuilt Loading Screen",
+  "quilt-loading-screen.midnightconfig.prideQuiltsEnabled": "Enable Pride Quilts even when it's not June!"
+}
```
Comment (sylv256) [inline src/main/resources/assets/quilt-loading-screen/lang/en_us.json:4]: newline expected at end of file

**Review by sylv256 — CHANGES_REQUESTED** [review]

File: src/main/resources/assets/quilt-loading-screen/lang/en_us.json (hunk 2 above)
Comment (triphora) [reply src/main/resources/assets/quilt-loading-screen/lang/en_us.json:4]: Doesn't really matter, json files work just fine without one

File: src/main/java/io/github/darkerbit/quiltloadingscreen/QuiltLoadingScreen.java [hunk 3]
```diff
@@ -35,13 +37,17 @@
...
+    private boolean prideMonth;
 
     public QuiltLoadingScreen(MinecraftClient client) {
         this.client = client;
 
         prideMonth = LocalDate.now().getMonth() == Month.JUNE;
 
+        Config.initConfig();
+
+        if (Config.isPrideQuiltsEnabled()) prideMonth = true;
```
Comment (halotroop2288) [inline src/main/java/io/github/darkerbit/quiltloadingscreen/QuiltLoadingScreen.java:49]: *`if (true) prideMonth = true;`

Pride month never ends.

**Review by halotroop2288 — APPROVED** [review]

**Comment by triphora** [comment]
> Alright, I'll take a closer look in the morning then! +1

two mornings later...

**Comment by darkerbit** [comment]
Finally getting back to this, gonna test it with the modmenu version from Quilt's maven since Terraformers maven is still down.

**Comment by darkerbit** [comment]
Alright, I've just tested it out a bit and have identified a problem: It has a hard dependency on Modmenu, and I don't want that

**Comment by triphora** [comment]
> Alright, I've just tested it out a bit and have identified a problem: It has a hard dependency on Modmenu, and I don't want that

What do you mean by that? It loads just fine for me without Mod Menu, and the config is created as well; the thing is that you won't be able to configure it from in-game if you don't have Mod Menu.

**Comment by darkerbit** [comment]
Isn't it convenient when a bunch of stuff happens that distracts me from a PR? Anyway, I've tried it out again and I have no idea what I was referring to with "hard dependency on Modmenu" and it works just fine without. I've moved MidnightConfig into the mod's package tree because if another mod has a MidnightConfig on that same package it might conflict. I've just finished the 1.17 version and I'll see if I can put it on 1.16 as well

**Comment by darkerbit** [comment]
Alright, I don't think I'll be able to put it on 1.16 soon, but the 1.17 version is done and I'm about to push it (and have tested it in 1.17.1 as well)

**Pull request merged**